### PR TITLE
Fix filters in Case List Report

### PR DESCRIPTION
**Pull request by AmitPhulera**
## Summary

https://dimagi-dev.atlassian.net/browse/SAAS-11658
## Product Description

If a user id was selected with any filter in Case List Reports the user ids were ignored, The PR fixes that.
This is a follow-up of https://github.com/dimagi/commcare-hq/pull/29015 in which we fixed a similar issue in Submission History Reports.
## Safety Assurance

- [x] Risk label is set correctly
- [x] All migrations are backward compatible and won't block deploy
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change
- [x] If QA is part of the safety story, the "Awaiting QA" label is used
- [x] I have confidence that this PR will not introduce a regression for the reasons below

### Automated test coverage

Tests for the changes have been added with the PR.
### QA Plan
https://dimagi-dev.atlassian.net/browse/QA-2424

### Safety story

I have tested the changes on local and will be deploying it to staging for QA.
### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### CODE REVIEW

File: corehq/apps/reports/standard/cases/basic.py [hunk 1]
```diff
@@ -65,25 +65,30 @@ def _build_query(self):
         if self.case_status:
             query = query.is_closed(self.case_status == 'closed')
 
-        if self.request.can_access_all_locations and (
-                EMWF.show_all_data(mobile_user_and_group_slugs)
-                or EMWF.no_filters_selected(mobile_user_and_group_slugs)
-        ):
-            pass
+        case_onwer_filters = []
```
Comment (proteusvacuum) [inline corehq/apps/reports/standard/cases/basic.py:68]: I think this is a spelling error, right? 
```suggestion
        case_owner_filters = []
```

File: corehq/apps/reports/tests/test_case_list_report.py [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+
+    def test_with_deactivated_slug(self):
+        report_slugs = ['t__5']
+        q_dict_get = QueryDict('', mutable=True)
+        q_dict_get.setlist('case_list_filter', report_slugs)
+        self.request.GET = q_dict_get
+        data = CaseListReport(self.request, domain=self.domain).es_results['hits'].get('hits', [])
+        expected_case_ids = ['id-1']
+        queried_case_ids = [case['_id'] for case in data]
+        sorted(queried_case_ids)
```
Comment (proteusvacuum) [inline corehq/apps/reports/tests/test_case_list_report.py:108]: out of curiosity, why do you sort these? I think you can use `assertItemsEqual` to check that two lists have the same things in them.

**Review by proteusvacuum — APPROVED** [review]
Looks good! A few comments.

File: corehq/apps/reports/tests/test_case_list_report.py (hunk 2 above)
Comment (AmitPhulera) [reply corehq/apps/reports/tests/test_case_list_report.py:108]: I guess just for sanity 😅  
`assertItemEqual` has been replaced by `assertCountEqual` in Python 3, I have used that here and I agree removing sorted will be a good idea here.
Thanks for pointing it out.

File: corehq/apps/reports/standard/cases/basic.py (hunk 1 above)
Comment (AmitPhulera) [reply corehq/apps/reports/standard/cases/basic.py:68]: Yes. It is a typo. Thanks for catching it 😄

File: corehq/apps/reports/tests/test_case_list_report.py (hunk 2 above)
Comment (proteusvacuum) [reply corehq/apps/reports/tests/test_case_list_report.py:108]: Whoa, that's a terrible name for that method 😄 
https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertCountEqual

>Test that sequence first contains the same elements as second, regardless of their order.

Comment (AmitPhulera) [reply corehq/apps/reports/tests/test_case_list_report.py]: Done

File: corehq/apps/reports/standard/cases/basic.py (hunk 1 above)
Comment (AmitPhulera) [reply corehq/apps/reports/standard/cases/basic.py]: Fixed.

File: corehq/apps/reports/tests/test_case_list_report.py (hunk 2 above)
Comment (AmitPhulera) [reply corehq/apps/reports/tests/test_case_list_report.py]: > Whoa, that's a terrible name for that method 😄

Yeah. I was confused at first and thought it has something to do with the count of elements.

File: corehq/apps/reports/standard/cases/basic.py [hunk 3]
```diff
@@ -65,25 +65,30 @@ def _build_query(self):
...
-              and EMWF.show_deactivated_data(mobile_user_and_group_slugs)):
-            query = query_deactivated_data(query, self.domain)
+        if (
+            self.request.can_access_all_locations
+            and EMWF.show_deactivated_data(mobile_user_and_group_slugs)
+        ):
+            case_owner_filters.append(deactivated_case_owners(self.domain))
 
-        else:  # Only show explicit matches
+        # Only show explicit matches
```
Comment (esoergel) [inline corehq/apps/reports/standard/cases/basic.py:82]: I think this comment is now innaccurate.  That is, the report will show explicit matches in addition to other matches.

File: corehq/apps/reports/standard/cases/basic.py [hunk 4]
```diff
@@ -65,25 +65,30 @@ def _build_query(self):
...
+        if (
+            self.request.can_access_all_locations
+            and EMWF.show_project_data(mobile_user_and_group_slugs)
+        ):
+            case_owner_filters.append(all_project_data_filter(self.domain, mobile_user_and_group_slugs))
 
-        elif (self.request.can_access_all_locations
-              and EMWF.show_deactivated_data(mobile_user_and_group_slugs)):
-            query = query_deactivated_data(query, self.domain)
+        if (
```
Comment (esoergel) [inline corehq/apps/reports/standard/cases/basic.py:76]: Should this still be an `elif`?  Isn't `deactivated_case_owners` a strict subset of `all_project_data`?

**Review by esoergel — COMMENTED** [review]
Sorry for the late review, just came across this

**Pull request merged**